src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
You're right.  The current code is WRONG.  I glossed over that.  The Windows version of HttpClientHandler (HttpClientHandler.Windows.cs) uses a slightly different way of expressing this since it has to do lazy setting of this operations and validates only in the .SendAsync() method.

But my comment still stands:

>It is not logically equivalent. If the value is CurlProxyUsePolicy.UseDefaultProxy, then this will return FALSE. But that is not correct. The HttpClientHandler.UseProxy Boolean should be true whether it is a 'default' proxy or a 'custom' proxy which uses the Proxy property IWebProxy. The only time that HttpClientHandler.UseProxy should be FALSE is if the policy is .DoNotUseProxy.